tasks/secure-installation.yml

```diff
@@ -1,87 +1,73 @@
+  loop: "{{ _mariadb_anonymous_hosts.query_result | first }}"
+  when: _mariadb_anonymous_hosts.rowcount[0] != 0
 
 - name: Get list of hosts for the root user.
-  ansible.builtin.command: mysql -NBe
-    "SELECT Host FROM mysql.user
-    WHERE User = '{{ mariadb_admin_user }}'
-    ORDER BY (Host='localhost') ASC"
+  community.mysql.mysql_query:
+    login_db: "mysql"
+    login_unix_socket: "{{ _mariadb_socket }}"
+    query: SELECT Host FROM mysql.user WHERE User = %s ORDER BY (Host='localhost') ASC
+    positional_args:
+      - "{{ mariadb_admin_user }}"
   register: _mariadb_root_hosts
-  changed_when: false
   check_mode: false
-  when: mariadb_admin_force_password_update
-
-# Note: We do not use mariadb_user for this operation, as it doesn't always update
-# the root password correctly. See: https://goo.gl/MSOejW
-# Set root password for MySQL >= 5.7.x.
-- name: Update MariaDB root password for localhost root account (5.7.x).
-  ansible.builtin.command: mysql -u root -NBe
-    'ALTER USER "{{ mariadb_admin_user }}"@"{{ item }}"
-    IDENTIFIED WITH mariadb_native_password BY "{{ mariadb_admin_password }}"; FLUSH PRIVILEGES;'
-  changed_when: false
-  loop: "{{ _mariadb_root_hosts.stdout_lines | d([]) }}"
-  register: _mariadb_root_password_update
-  no_log: "{{ not mariadb_debug }}"
-  when: >
-    mariadb_admin_force_password_update and
-    ('5.7.' in _mariadb_cli_version_.stdout or '8.0.' in _mariadb_cli_version_.stdout)
 
-# Set root password for MySQL < 5.7.x.
-- name: Update MariaDB root password for localhost root account (< 5.7.x).
-  ansible.builtin.command: mysql -NBe
-    'SET PASSWORD FOR "{{ mariadb_admin_user }}"@"{{ item }}" = PASSWORD("{{ mariadb_admin_password }}"); FLUSH PRIVILEGES;'
-  loop: "{{ _mariadb_root_hosts.stdout_lines | d([]) }}"
+- name: Update MariaDB root password for localhost root account
+  community.mysql.mysql_user:
+    name: "{{ mariadb_admin_user }}"
+    password: "{{ mariadb_admin_password }}"
+    login_unix_socket: "{{ _mariadb_socket }}"
+    host: "{{ item.Host }}"
+  loop: "{{ _mariadb_root_hosts.query_result | first }}"
   register: _mariadb_root_password_update
-  changed_when: false
-  no_log: "{{ not mariadb_debug }}"
-  when: >
-    mariadb_admin_force_password_update and
-    ('5.7.' not in _mariadb_cli_version_.stdout and '8.0.' not in _mariadb_cli_version_.stdout)
+  when:
+    - _mariadb_root_hosts.rowcount[0] != 0
+    - mariadb_admin_force_password_update
 
 - name: Output results of updating mysql root password
   ansible.builtin.debug:
     var: _mariadb_root_password_update
   when: mariadb_debug
 
-# Has to be after the root password assignment, for idempotency.
-- name: Copy .my.cnf file with root password credentials.
-  ansible.builtin.template:
-    src: "root-my.cnf.j2"
-    dest: "{{ mariadb_admin_home }}/.my.cnf"
-    owner: root
-    group: root
-    mode: 0600
-  when: mariadb_admin_force_password_update
-
 - name: Remove MariaDB test database.
```

Comment:
This task is required to ease login when connecting to the database as the admin (root) user. Otherwise one has to provide the password everytime in order to login as the root user and that is generally very tedious.
This opinion is still debatable though.